Doc/howto/clinic.rst

```diff
@@ -93,67 +96,143 @@ and it should be able to do many interesting and smart
-All of these lines, including the beginning and ending comment
-lines, are collectively called an Argument Clinic "block".
-
-When Argument Clinic parses one of these blocks, it
-generates output.  This output is rewritten into the C file
-immediately after the block, followed by a comment containing a checksum.
-The Argument Clinic block now looks like this:
+Everything in between these two lines is Argument Clinic :term:`input`.
+When Argument Clinic parses input, it generates :term:`output`.
+The output is rewritten into the C file immediately after the input,
+followed by a :term:`checksum line`.
+All of these lines, including the :term:`start line` and :term:`checksum line`,
+are collectively called an Argument Clinic :term:`block`:
 
-.. code-block:: none
+.. code-block:: c
 
     /*[clinic input]
     ... clinic input goes here ...
     [clinic start generated code]*/
     ... clinic output goes here ...
-    /*[clinic end generated code: checksum=...]*/
+    /*[clinic end generated code: ...]*/
 
 If you run Argument Clinic on the same file a second time, Argument Clinic
-will discard the old output and write out the new output with a fresh checksum
-line.  However, if the input hasn't changed, the output won't change either.
-
-You should never modify the output portion of an Argument Clinic block.  Instead,
-change the input until it produces the output you want.  (That's the purpose of the
-checksum—to detect if someone changed the output, as these edits would be lost
-the next time Argument Clinic writes out fresh output.)
-
-For the sake of clarity, here's the terminology we'll use with Argument Clinic:
-
-* The first line of the comment (``/*[clinic input]``) is the *start line*.
-* The last line of the initial comment (``[clinic start generated code]*/``) is the *end line*.
-* The last line (``/*[clinic end generated code: checksum=...]*/``) is the *checksum line*.
-* In between the start line and the end line is the *input*.
-* In between the end line and the checksum line is the *output*.
-* All the text collectively, from the start line to the checksum line inclusively,
-  is the *block*.  (A block that hasn't been successfully processed by Argument
-  Clinic yet doesn't have output or a checksum line, but it's still considered
-  a block.)
+will discard the old :term:`output` and write out the new output with a fresh
+:term:`checksum line`.
+If the :term:`input` hasn't changed, the output won't change either.
+
+.. note::
+
+   You should never modify the output of an Argument Clinic block,
+   as any change will be lost in future Argument Clinic runs;
+   Argument Clinic willl detect an output checksum mismatch and regenerate the
+   correct output.
+   If you are not happy with the generated output,
+   change instead the input until it produces the output you want.
+
+
+.. _clinic-reference:
+
+Reference
+=========
+
+
+.. _clinic-terminology:
+
+Terminology
+-----------
+
+.. glossary::
```

Comment:
This is unfortunately global; I'm not sure if it is possible to make it file local. @AA-Turner?

Anyway, _currently_, there's no collision between the _Glossary_, and the Argument Clinic glossary. I'm not sure if that's good enough though.

Alternatively, we can just mark this up with something else, but the `:term:` directives are nice to have...

Reply:
Namespaces for glossaries in Sphinx need further design work -- it is a known deficiency currently.

You could use *start line (argument clinic)* as the term, but that is uglier.

For context, see https://github.com/sphinx-doc/sphinx/issues/1399 and https://github.com/sphinx-doc/sphinx/pull/10517.

A

Reply:
Thanks for the info, Adam! If the docs folks are ok with it, we can keep the Argument Clinic glossary for now, and cross our fingers that there won't be any collisions until Sphinx gets better support for multiple glossaries :)

Reply:
See 72fe212adb9b87609ac5ca1232472e87acb749bf